src/init.rs

```diff
@@ -24,30 +33,254 @@ description = ""
+        );
+        manifest.package.name = ask_until_valid(
+            "Package name",
+            Some(manifest.package.name),
+            util::validate_name,
+        )?;
+        manifest.package.version = ask_until_valid(
+            "Version",
+            Some(manifest.package.version.to_string()),
+            Version::parse,
+        )?;
+        manifest.package.description =
+            ask("Description", Some(manifest.package.description))?.unwrap_or("".to_owned());
+        manifest.package.repository = ask("Repository", manifest.package.repository)?;
+        // author = ask("Author", &author)?;
+        manifest.package.license = Some(ask_until_valid(
+            "License",
+            manifest.package.license,
+            util::validate_license,
+        )?);
+        // Let's reset the modules
+        let mut all_modules: Vec<Module> = vec![];
+        let mut all_commands: Vec<Command> = vec![];
+        loop {
+            let current_index = all_modules.len();
+            println!("Enter the data for the Module ({})", current_index + 1);
+            let mut module = if current_index == 0 {
+                Module {
+                    name: "entry".to_owned(),
+                    source: PathBuf::from("entry.wasm"),
+                    abi: Abi::default(),
+                    interfaces: None,
+                }
+            } else {
+                Module {
+                    name: "".to_owned(),
+                    source: PathBuf::from("none"),
+                    abi: Abi::default(),
+                    interfaces: None,
+                }
+            };
+            module.source = ask_until_valid(
+                " - Source (path)",
+                Some(module.source.to_str().unwrap().to_owned()),
+                validate_wasm_source,
+            )?;
+            if module.source.to_str().unwrap() == "none" {
+                break;
+            }
+            // Let's try to guess the name based on the file path
+            let default_module_name = Path::new(&module.source)
+                .file_stem()
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .to_owned();
+            module.name = ask_until_valid(
+                " - Name",
+                Some(default_module_name.clone()),
+                util::validate_name,
+            )?;
+            let (abi, interfaces): (Abi, Option<HashMap<String, String>>) = match Select::new()
+                .with_prompt(" - ABI")
+                .item("None")
+                .item("WASI")
+                .item("Emscripten")
+                .default(0)
+                .interact()?
+            {
+                1 => (
+                    Abi::Wasi,
+                    Some(
+                        [("wasi".to_owned(), WASI_LAST_VERSION.to_owned())]
+                            .iter()
+                            .cloned()
+                            .collect(),
```

Comment:
⛳️ 
```Rust
.into_iter().collect()
```

Reply:
I tried this, but it doesn't work 🤷‍♂️

This code is transforming a list of tuples into a HashMap.

Reply:
Interesting, it's because it's a slice I guess 🤔